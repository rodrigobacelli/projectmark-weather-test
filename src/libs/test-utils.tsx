import * as React from 'react';
import { render, RenderOptions } from '@testing-library/react';

const Providers = ({ children }: { children: React.ReactNode }) => {
  return children;
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { customRender as render };
