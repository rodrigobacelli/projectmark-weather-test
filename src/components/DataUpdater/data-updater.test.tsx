import { describe, it, expect, vi } from 'vitest';
import { render } from '../../tests/test-utils';
import { DataUpdater } from './';
import { format } from 'date-fns/format';

const commonDate = new Date(1740798000);

describe('components/DataUpdater', () => {
  it('should render the date label when date is set', () => {
    const result = render(<DataUpdater lastUpdated={commonDate} />);

    expect(result.container.getElementsByTagName('time')[0]).toHaveAttribute(
      'dateTime',
      format(commonDate, 'yyyy-MM-dd HH:mm:ss')
    );
  });

  it('should render the DataUpdater update button', () => {
    const result = render(<DataUpdater />);

    expect(result.getByRole('button')).toBeInTheDocument();
    expect(result.getByRole('button')).toContainHTML('Update');
  });

  it('should handle DataUpdater update button click', async () => {
    const onUpdate = vi.fn();
    const result = render(<DataUpdater onUpdate={onUpdate} />);

    await result.getByRole('button').click();

    expect(onUpdate).toHaveBeenCalledTimes(1);
  });

  it('should disable the update button when isUpdating is true', () => {
    const result = render(<DataUpdater isUpdating />);

    expect(result.getByRole('button')).toBeDisabled();
  });

  it('should accept classNames', () => {
    const result = render(<DataUpdater className="data-updater-className" />);

    expect(
      result.container.querySelector('.data-updater-className')
    ).toBeInTheDocument();
  });

  it('should render the DataUpdater Component', () => {
    const result = render(
      <DataUpdater
        lastUpdated={commonDate}
        className="data-updated-className"
      />
    );

    expect(result).toMatchSnapshot();
  });
});
