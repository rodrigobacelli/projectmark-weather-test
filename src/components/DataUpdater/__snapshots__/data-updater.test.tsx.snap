// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/DataUpdater > should render the DataUpdater Component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex flex-col-reverse md:flex-row gap-2 justify-end items-end md:items-center data-updated-className"
      >
        <p
          class="text-gray-900 dark:text-gray-100 text-xs"
        >
          Updated at
           
          <time
            datetime="1970-01-21 00:33:18"
          >
            01/21/1970 00:33:18
          </time>
        </p>
        <button
          class="text-sm flex items-center gap-1 px-2 py-1 rounded-sm border enabled:cursor-pointer text-gray-900 dark:text-gray-300 border-gray-900 dark:border-gray-300 enabled:hover:bg-gray-900/20 enabled:dark:hover:bg-gray-100/20 enabled:focus:bg-gray-800/10 enabled:dark:focus:bg-gray-200/10 disabled:text-gray-500 disabled:border-gray-500"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 15 15"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M1.90321 7.29677C1.90321 10.341 4.11041 12.4147 6.58893 12.8439C6.87255 12.893 7.06266 13.1627 7.01355 13.4464C6.96444 13.73 6.69471 13.9201 6.41109 13.871C3.49942 13.3668 0.86084 10.9127 0.86084 7.29677C0.860839 5.76009 1.55996 4.55245 2.37639 3.63377C2.96124 2.97568 3.63034 2.44135 4.16846 2.03202L2.53205 2.03202C2.25591 2.03202 2.03205 1.80816 2.03205 1.53202C2.03205 1.25588 2.25591 1.03202 2.53205 1.03202L5.53205 1.03202C5.80819 1.03202 6.03205 1.25588 6.03205 1.53202L6.03205 4.53202C6.03205 4.80816 5.80819 5.03202 5.53205 5.03202C5.25591 5.03202 5.03205 4.80816 5.03205 4.53202L5.03205 2.68645L5.03054 2.68759L5.03045 2.68766L5.03044 2.68767L5.03043 2.68767C4.45896 3.11868 3.76059 3.64538 3.15554 4.3262C2.44102 5.13021 1.90321 6.10154 1.90321 7.29677ZM13.0109 7.70321C13.0109 4.69115 10.8505 2.6296 8.40384 2.17029C8.12093 2.11718 7.93465 1.84479 7.98776 1.56188C8.04087 1.27898 8.31326 1.0927 8.59616 1.14581C11.4704 1.68541 14.0532 4.12605 14.0532 7.70321C14.0532 9.23988 13.3541 10.4475 12.5377 11.3662C11.9528 12.0243 11.2837 12.5586 10.7456 12.968L12.3821 12.968C12.6582 12.968 12.8821 13.1918 12.8821 13.468C12.8821 13.7441 12.6582 13.968 12.3821 13.968L9.38205 13.968C9.10591 13.968 8.88205 13.7441 8.88205 13.468L8.88205 10.468C8.88205 10.1918 9.10591 9.96796 9.38205 9.96796C9.65819 9.96796 9.88205 10.1918 9.88205 10.468L9.88205 12.3135L9.88362 12.3123C10.4551 11.8813 11.1535 11.3546 11.7585 10.6738C12.4731 9.86976 13.0109 8.89844 13.0109 7.70321Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          Update
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex flex-col-reverse md:flex-row gap-2 justify-end items-end md:items-center data-updated-className"
    >
      <p
        class="text-gray-900 dark:text-gray-100 text-xs"
      >
        Updated at
         
        <time
          datetime="1970-01-21 00:33:18"
        >
          01/21/1970 00:33:18
        </time>
      </p>
      <button
        class="text-sm flex items-center gap-1 px-2 py-1 rounded-sm border enabled:cursor-pointer text-gray-900 dark:text-gray-300 border-gray-900 dark:border-gray-300 enabled:hover:bg-gray-900/20 enabled:dark:hover:bg-gray-100/20 enabled:focus:bg-gray-800/10 enabled:dark:focus:bg-gray-200/10 disabled:text-gray-500 disabled:border-gray-500"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 15 15"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M1.90321 7.29677C1.90321 10.341 4.11041 12.4147 6.58893 12.8439C6.87255 12.893 7.06266 13.1627 7.01355 13.4464C6.96444 13.73 6.69471 13.9201 6.41109 13.871C3.49942 13.3668 0.86084 10.9127 0.86084 7.29677C0.860839 5.76009 1.55996 4.55245 2.37639 3.63377C2.96124 2.97568 3.63034 2.44135 4.16846 2.03202L2.53205 2.03202C2.25591 2.03202 2.03205 1.80816 2.03205 1.53202C2.03205 1.25588 2.25591 1.03202 2.53205 1.03202L5.53205 1.03202C5.80819 1.03202 6.03205 1.25588 6.03205 1.53202L6.03205 4.53202C6.03205 4.80816 5.80819 5.03202 5.53205 5.03202C5.25591 5.03202 5.03205 4.80816 5.03205 4.53202L5.03205 2.68645L5.03054 2.68759L5.03045 2.68766L5.03044 2.68767L5.03043 2.68767C4.45896 3.11868 3.76059 3.64538 3.15554 4.3262C2.44102 5.13021 1.90321 6.10154 1.90321 7.29677ZM13.0109 7.70321C13.0109 4.69115 10.8505 2.6296 8.40384 2.17029C8.12093 2.11718 7.93465 1.84479 7.98776 1.56188C8.04087 1.27898 8.31326 1.0927 8.59616 1.14581C11.4704 1.68541 14.0532 4.12605 14.0532 7.70321C14.0532 9.23988 13.3541 10.4475 12.5377 11.3662C11.9528 12.0243 11.2837 12.5586 10.7456 12.968L12.3821 12.968C12.6582 12.968 12.8821 13.1918 12.8821 13.468C12.8821 13.7441 12.6582 13.968 12.3821 13.968L9.38205 13.968C9.10591 13.968 8.88205 13.7441 8.88205 13.468L8.88205 10.468C8.88205 10.1918 9.10591 9.96796 9.38205 9.96796C9.65819 9.96796 9.88205 10.1918 9.88205 10.468L9.88205 12.3135L9.88362 12.3123C10.4551 11.8813 11.1535 11.3546 11.7585 10.6738C12.4731 9.86976 13.0109 8.89844 13.0109 7.70321Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        Update
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
